### ISSUE TITLE
ClusterClass quickstart e2e test fails with CAPI v1.1.0-beta.1 update

### ISSUE THREAD

**Issue by invidian** [kubernetes-sigs/cluster-api-provider-aws#3054]
/kind bug

**What steps did you take and what happened:**
Running e2e tests as part of #2271, e2e tests sometimes fails with unrelated error. See https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/2271/pull-cluster-api-provider-aws-e2e/1478140086141849600.

**What did you expect to happen:**

Untouched e2e tests to pass on every run.

**Comment by sedefsavas** [comment]
I don't see any flakiness in the periodic tests:
https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-aws#periodic-e2e-main&show-stale-tests=

**Comment by invidian** [comment]
There error in the job is the following:
```
[2] error: error validating "STDIN": error validating data: [ValidationError(KubeadmControlPlaneTemplate.spec.template.spec.machineTemplate): unknown field "infrastructureRef" in io.x-k8s.cluster.controlplane.v1beta1.KubeadmControlPlaneTemplate.spec.template.spec.machineTemplate, ValidationError(KubeadmControlPlaneTemplate.spec.template.spec): unknown field "replicas" in io.x-k8s.cluster.controlplane.v1beta1.KubeadmControlPlaneTemplate.spec.template.spec, ValidationError(KubeadmControlPlaneTemplate.spec.template.spec): unknown field "version" in io.x-k8s.cluster.controlplane.v1beta1.KubeadmControlPlaneTemplate.spec.template.spec]; if you choose to ignore these errors, turn validation off with --validate=false
```

It might be related to my updates to CAPI v1.1.0-beta1 and nightlies, or maybe it's some sort of race related to #3057, as it seems I'm able to reproduce it quite reliably right now. It seems it is still worth investigating, as the error message looks rather bizarre.

**Comment by sedefsavas** [comment]
Let's wait to see if this happens when using consistent CAPI version with manifests + binary.
Nightly build we use and the beta release are not necessarily compatible.

**Comment by invidian** [comment]
It looks like this is related to CAPI v1.1.0-beta.1 update, hitting in #3058 as well.

**Comment by invidian** [comment]
Pushed fixes in #2271 where tests are passing now, this PR includes #3058 changes, so feel free to pull my fixes there, if applicable. I'll close this one.
